Servers/UI/OJS.Servers.Ui/ClientApp/src/components/profile/profile-contest-participations/ProfileContestParticipations.tsx

```diff
@@ -1,106 +1,142 @@
-    },
-    {
-        field: 'competeResult',
-        headerName: 'Compete Result',
-        type: 'string',
-        minWidth: 70,
-        flex: 1,
-        sortable: false,
-        valueGetter: (params: GridValueGetterParams) => `${params.row.competeResult}/${params.row.contestCompeteMaximumPoints}`,
-    },
-    {
-        field: 'practiceResult',
-        headerName: 'Practice Result',
-        type: 'string',
-        minWidth: 70,
-        flex: 1,
-        sortable: false,
-        valueGetter: (params: GridValueGetterParams) => `${params.row.practiceResult}/${params.row.contestPracticeMaximumPoints}`,
-    },
-];
-
-const ProfileContestParticipations = () => {
-    const [ numberedRows, setNumberedRows ] =
-        useState<Array<IParticipationType>>([]);
-    const { internalUser: user } =
-        useSelector((state: {authorization: IAuthorizationReduxState}) => state.authorization);
+import { SortType, SortTypeDirection } from '../../../common/contest-types';
+import { IGetContestParticipationsForUserQueryParams, IIndexContestsType } from '../../../common/types';
+import useTheme from '../../../hooks/use-theme';
+import {
+    setProfileUserContestParticipationsPage,
+    setUserContestParticipations,
+} from '../../../redux/features/contestsSlice';
+import { useGetContestsParticipationsForUserQuery } from '../../../redux/services/contestsService';
+import { useAppDispatch, useAppSelector } from '../../../redux/store';
+import concatClassNames from '../../../utils/class-names';
+import ContestCard from '../../contests/contest-card/ContestCard';
+import List, { Orientation } from '../../guidelines/lists/List';
+import PaginationControls from '../../guidelines/pagination/PaginationControls';
+import SpinningLoader from '../../guidelines/spinning-loader/SpinningLoader';
+
+import styles from './ProfileContestParticipations.module.scss';
+
+interface IProfileContestParticipationsProps {
+    userIsProfileOwner: boolean;
+    isChosenInToggle: boolean;
+}
+
+const ProfileContestParticipations = ({ userIsProfileOwner, isChosenInToggle }: IProfileContestParticipationsProps) => {
+    const [ shouldSkipFetchData, setShouldSkipFetchData ] = useState<boolean>(true);
+    const [ shouldRender, setShouldRender ] = useState<boolean>(false);
+
+    const { internalUser, isLoggedIn } = useAppSelector((reduxState) => reduxState.authorization);
+    const { profile } = useAppSelector((reduxState) => reduxState.users);
     const {
-        state: { userParticipations },
-        actions: { getUserParticipations },
-    } = useParticipations();
-    const { state: { isProfileInfoLoaded, myProfile } } = useUsers();
-    const { username } = useParams();
+        userContestParticipations,
+        profileUserContestParticipationsPage,
+    } = useAppSelector((reduxState) => reduxState.contests);
+    const { getColorClassName, themeColors } = useTheme();
+
+    const dispatch = useAppDispatch();
+
+    const {
+        data: contestsParticipations,
```

Comment:
dont we need to handle the error here as well?

Reply:
Error?

Reply:
from the fetch, it has data, isLoading and error, for when the API response has failed, we can use the error in order to show the user proper message

Reply:
Fixed